x-pack/plugins/enterprise_search/server/routes/app_search/app_search_request_handler.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { mockConfig, mockLogger } from '../__mocks__';
+
+import { createAppSearchRequestHandler } from './app_search_request_handler';
+
+jest.mock('node-fetch');
+const fetch = jest.requireActual('node-fetch');
+const { Response } = fetch;
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fetchMock = require('node-fetch') as jest.Mocked<typeof fetch>;
+
+const responseMock = {
+  ok: jest.fn(),
+  notFound: jest.fn(),
+};
+
+beforeEach(() => {
+  responseMock.ok.mockClear();
+  responseMock.notFound.mockClear();
+  fetchMock.mockReset();
+});
+
+describe('createAppSearchRequestHandler', () => {
+  it('makes an API call and returns the response', async () => {
+    const appSearchAPIResponseBody = {
+      results: [{ name: 'engine1' }],
+      meta: { page: { total_results: 1 } },
+    };
+    const KibanaAuthHeader = 'Basic 123';
```

Comment:
Can we pull `KibanaAuthHeader` out to a top-level constant (near `const responseMock` perhaps?) since we're repeating it in multiple tests?

I'd prefer DRYing out the full thing, i.e.
`const headers = { authorization: 'Basic 123' }`
so that `shouldHaveBeenCalledWith` can DRY out & use a `{ headers }` arg as well